project/apps/matching-service/src/matchEngine/matchEngine.consumeRequest.ts

```diff
@@ -25,21 +26,26 @@ export class MatchEngineConsumer implements OnModuleInit {
         await channel.assertQueue(MATCH_QUEUE, { durable: true });
         await channel.consume(MATCH_QUEUE, async (message) => {
           if (message) {
-            const content = JSON.parse(message.content.toString());
-            this.consumeMessage(content);
-            channel.ack(message);
+            try {
+              const content = JSON.parse(message.content.toString());
+              const matchRequest: MatchRequestMsgDto = matchRequestMsgSchema.parse(content);
```

Comment:
typescript will know its typing after the parse
```suggestion
              const matchRequest = matchRequestMsgSchema.parse(content);
```